torchtune/modules/peft/lora.py

```diff
@@ -128,6 +134,17 @@ def forward(self, x: Tensor) -> Tensor:
             return out
         lora_out = self.lora_a(self.dropout(x))
         lora_out = (self.alpha / self.rank) * self.lora_b(lora_out)
+        # Author mentions this method is faster for the computation purpose:
+        # https://github.com/huggingface/peft/pull/1474#issuecomment-1963402710
+        if self.use_dora:
+            weight_norm = torch.linalg.norm(
+                self.weight
+                + (self.alpha / self.rank)
+                * (self.lora_a.weight.T @ self.lora_b.weight.T).T,
+                dim=1,
+            ).to(self.weight.dtype)
+            mag_norm_scale = (self.m / weight_norm - 1).view(1, -1)
+            return mag_norm_scale * out + mag_norm_scale * lora_out
```

Comment:
Sorry maybe I am still missing the point but I don't think this actually matches the version that was ultimately added to PEFT. [Ref](https://github.com/huggingface/peft/blob/02ae6bcb373d9d9d3bec9ba920d63316418ff64a/src/peft/tuners/lora/layer.py#L250-L252)